ch4.전역 스코프에 대해서

YeonjuOHYE opened the pull request.

YeonjuOHYE committed:
ch4

YeonjuOHYE committed:
Update ch4.md

YeonjuOHYE committed:
Update ch4.md

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -1,25 +1,25 @@
+# 4장: 전역 스코프<sub>Global Scope</sub>에 대하여
 
-Chapter 3 mentioned the "global scope" several times, but you may still be wondering why a program's outermost scope is all that important in modern JS. The vast majority of work is now done inside of functions and modules rather than globally.
+3장에서는 "전역 스코프"에 대해 여러 번 언급했지만, 여전히 모던 JS에서 프로그램의 가장 바깥쪽 범위가 왜 그렇게 중요한지 궁금할 수 있다. 이제는 대부분의 작업이 전역이 아닌 함수 및 모듈 내부에서 이루어고 있다.
```
`outermost scope`에서 `가장 바깥쪽 스코프`라고 그대로 해석해도 될 것 같아요.

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -1,25 +1,25 @@
+렉시컬 스코프를 사용하여 프로그램을 구성하는 데 있어 전역 스코프를 완전히 이해하는 것은 매우 중요하다.
 
-## Why Global Scope?
+## 왜 전역 스코프가 중요할까?
```
👍

01kn replied:
👍

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -39,13 +39,13 @@ In some build setups, the entire contents of the file are wrapped in a single en
+`wrappingOuterScope()`의 범위는 전체 환경 전역 스코프가 아니라 함수이지만, 최상위 식별자를 모두 저장할 수 있는 버킷으로 작동하며(실제로는 전역 스코프에 저장하는 것은 아니지만) "응용 프로그램 스코프<sub>application-wide scope</sub>" 역할을 한다. 그런 점에서 전역 스코프를 대신하는 역할을 한다.
 
-And finally, the third way: whether a bundler tool is used for an application, or whether the (non-ES module) files are simply loaded in the browser individually (via `<script>` tags or other dynamic JS resource loading), if there is no single surrounding scope encompassing all these pieces, the **global scope** is the only way for them to cooperate with each other:
+마지막 세번째 방법으로, 번들러 도구가 애플리케이션에 사용되는 않고, (ES 모듈이 아닌) 파일이 단순히 브라우저에 개별적으로 로드가 되지 않아('스크립트' 태그 또는 기타 동적 JS 리소스 로딩을 통해) 이러한 모든 부분을 둘러싸는 하나의 스코프가 없으면  **전역 스코프*가 유일하게 연결할 수 있는 방법이다.
```
`사용되는 않고,` -> `사용되던지`
`개별적으로 로드가 되지 않아` -> `개별적으로 로드 되던지`
whether~, or whether~, 부분을 `~하던지`로 해석해도 좋을 것 같아요

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -86,32 +86,33 @@ var moduleTwo = (function two(){
+이러한 파일들이 브라우저 환경에서 일반 독립 실행형 .js 파일로 별도로 로드되는 경우, 각 파일들의 최상위 변수 선언이 전역 변수가 된다. 왜냐하면 JS엔진에서는 전역 스코프만이 별개의 파일들(-독립적인 프로그램들)이 공유하는 유일한 리소스이기 떄문이다.
 
-In addition to (potentially) accounting for where an application's code resides during runtime, and how each piece is able to access the other pieces to cooperate, the global scope is also where:
+실행 시간 동안 응용프로그램의 코드가 존재하는 위치와 각 작업물이 다른 작업에 액세스하여 협력할 수 있는 방법을 설명하는 것 외에도  전역 스코프는 다음에서도 찾아볼 수 있다:
```
`piece`가 위에서부터 계속 나오는데 이 부분들을 `모듈`로 해석하는건 어떨까요?

01kn replied:
모듈의 뜻은 맞는데, 진작에 module로 써도 될 껄 안쓴거 보면... 다른 어감을 준 것 같아요. `조각`?

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -1,25 +1,25 @@
+둘째, 빌드 프로세스에서 번들러를 사용하는 경우 일반적으로 브라우저와 JS 엔진으로 전송하기 전에 모든 파일이 연결되고, 그 결과 브라우저와 JS엔진은 하나의 큰 파일만 처리한다. 애플리케이션의 모든 부분이 단일 파일에 함께 위치하더라도 각 부분이 다른 부분에서 참조할 *이름*을(를) 등록하는 메커니즘과 해당 액세스가 가능하도록 하기 위한 기능이 필요하다.
 
-In some build setups, the entire contents of the file are wrapped in a single enclosing scope, such as a wrapper function, universal module (UMD—see Appendix A), etc. Each piece can register itself for access from other pieces by way of local variables in that shared scope. For example:
+일부 빌드 설정에서는 파일의 전체 내용이 wrapper 함수, 범용 모듈(UMD—부록 A 참조) 등과 같은 하나의 닫힌 스코프(enclosing scope)로 래핑된다. 각 부분은 해당 공유 범위에서 로컬 변수를 사용하여 다른 부분에서 액세스할 수 있도록 스스로를 등록할 수 있다. 예를 들어, 다음과 같다.
```
여기도 `공유 범위` -> `공유 스코프`로 해석해도 될 것 같습니다

01kn replied:
`액세스할 수 있도록` -> `접근할 수 있도록`

01kn replied:
`wrapper 함수` -> `감싸는 함수` 
`파일의 전체 내용이 ~ 래핑된다.` -> `파일의 전체 내용을 ~ 감싼다.`

01kn commented on scope-closures/ch4.md:
```diff
@@ -1,25 +1,25 @@
 # You Don't Know JS Yet: Scope & Closures - 2nd Edition
-# Chapter 4: Around the Global Scope
+# 4장: 전역 스코프<sub>Global Scope</sub>에 대하여
```
전역 스코프 살펴보기? 전역 스코프 둘러보기?

01kn commented on scope-closures/ch4.md:
```diff
@@ -39,13 +39,13 @@ In some build setups, the entire contents of the file are wrapped in a single en
+위와 같이 `wrappingOuterScope()` 기능 범위 내 `moduleOne`과 `moduleTwo` 지역 변수를 선언하여 서로 접근할 수 있도록 했다.
 
-While the scope of `wrappingOuterScope()` is a function and not the full environment global scope, it does act as a sort of "application-wide scope," a bucket where all the top-level identifiers can be stored, though not in the real global scope. It's kind of like a stand-in for the global scope in that respect.
+`wrappingOuterScope()`의 범위는 전체 환경 전역 스코프가 아니라 함수이지만, 최상위 식별자를 모두 저장할 수 있는 버킷으로 작동하며(실제로는 전역 스코프에 저장하는 것은 아니지만) "응용 프로그램 스코프<sub>application-wide scope</sub>" 역할을 한다. 그런 점에서 전역 스코프를 대신하는 역할을 한다.
```
`버킷` -> `양동이`

01kn commented on scope-closures/ch4.md:
```diff
@@ -86,32 +86,33 @@ var moduleTwo = (function two(){
+| : |
 | :--- |
-| Node also exposes several elements "globally," but they're technically not in the `global` scope: `require()`, `__dirname`, `module`, `URL`, and so on. |
+| 노드도 "전역적으로" 여러 요소를 노출하지만 기술적으로는 `전역(global)` 스코프에 있지는 않다. : `require()`, `__dirname`, `module`, `URL`, 등등. |
```
```
Node(Node.js)도 "전역적으로" 몇 가지 요소를 노출하지만, 엄밀히 따지면 `전역(global)` 스코프에 있지는 않다: `require()`, `__dirname`, `module`, `URL` 등.
```

01kn commented on scope-closures/ch4.md:
```diff
@@ -86,32 +86,33 @@ var moduleTwo = (function two(){
+| 노드도 "전역적으로" 여러 요소를 노출하지만 기술적으로는 `전역(global)` 스코프에 있지는 않다. : `require()`, `__dirname`, `module`, `URL`, 등등. |
 
-Most developers agree that the global scope shouldn't just be a dumping ground for every variable in your application. That's a mess of bugs just waiting to happen. But it's also undeniable that the global scope is an important *glue* for practically every JS application.
+대부분의 개발자들은 전역 스코프가 단순히 응용프로그램의 모든 변수에 대한 덤핑장이 되어서는 안 된다는 데 동의한다. 엉망진창의 버그들이 기다리고 있다. 그러나 전역 스코프가 실질적으로 모든 JS 애플리케이션에 중요한 *접착제*라는 사실도 부인할 수 없다.
```
```
대부분의 개발자는 전역 스코프가 단순히 애플리케이션에서 모든 변수를 떠넘기는 장소가 되어서는 안 된다는 데 동의한다. 엉만진창의 버그가 기다리고 있다. 그러나 전역 스코프가 실질적으로 모든 JS 애플리케이션에 중요한 *접착제*라는 사실도 부인할 수 없다.
```
- 복수형 제거
- 애플리케이션
- dumping ground

01kn commented on scope-closures/ch4.md:
```diff
@@ -86,32 +86,33 @@ var moduleTwo = (function two(){
 
-* JS exposes its built-ins:
+
+* JS가 제공하는 built-ins:
```
`JS는 아래 빌트인<sub>built-ins</sub>을 노출한다:`?

01kn commented on scope-closures/ch4.md:
```diff
@@ -86,32 +86,33 @@ var moduleTwo = (function two(){
     - friends of JS: `Intl`, `WebAssembly`
 
-* The environment hosting the JS engine exposes its own built-ins:
+* JS을 호스팅하는 환경에서 제공하는 built-ins:
```
`JS 엔진을 호스팅하는 환경은 아래 빌트인을 노출한다:`?

YeonjuOHYE committed:
리뷰 반영

YeonjuOHYE committed:
리뷰2

sw-wm commented on scope-closures/ch4.md:
```diff
@@ -39,13 +39,13 @@ In some build setups, the entire contents of the file are wrapped in a single en
+`wrappingOuterScope()`의 범위는 전체 환경 전역 스코프가 아니라 함수이지만, 최상위 식별자를 모두 저장할 수 있는 양동이로 작동하며(실제로는 전역 스코프에 저장하는 것은 아니지만) "애플리케이션 스코프<sub>application-wide scope</sub>" 역할을 한다. 그런 점에서 전역 스코프를 대신하는 역할을 한다.
 
-And finally, the third way: whether a bundler tool is used for an application, or whether the (non-ES module) files are simply loaded in the browser individually (via `<script>` tags or other dynamic JS resource loading), if there is no single surrounding scope encompassing all these pieces, the **global scope** is the only way for them to cooperate with each other:
+마지막 세번째 방법으로, 번들러 도구가 애플리케이션에 사용되는 않거나, (ES 모듈이 아닌) 파일이 단순히 브라우저에 개별적으로 로드가 되지 않는('스크립트' 태그 또는 기타 동적 JS 리소스 로딩을 통해)것과 상관없이 이러한 모든 부분을 둘러싸는 하나의 스코프가 없으면  **전역 스코프*가 유일하게 연결할 수 있는 방법이다.
```
```suggestion
마지막 세번째 방법으로, 번들러 도구가 애플리케이션에 사용되지 않거나, (ES 모듈이 아닌) 파일이 단순히 브라우저에 개별적으로 로드가 되지 않는('스크립트' 태그 또는 기타 동적 JS 리소스 로딩을 통해)것과 상관없이 이러한 모든 부분을 둘러싸는 하나의 스코프가 없으면 **전역 스코프**가 유일하게 연결할 수 있는 방법이다.
```

`*`하나 더요... 하핫

sw-wm commented on scope-closures/ch4.md:
```diff
@@ -39,13 +39,13 @@ In some build setups, the entire contents of the file are wrapped in a single en
+마지막 세번째 방법으로, 번들러 도구가 애플리케이션에 사용되는 않거나, (ES 모듈이 아닌) 파일이 단순히 브라우저에 개별적으로 로드가 되지 않는('스크립트' 태그 또는 기타 동적 JS 리소스 로딩을 통해)것과 상관없이 이러한 모든 부분을 둘러싸는 하나의 스코프가 없으면  **전역 스코프*가 유일하게 연결할 수 있는 방법이다.
 
-A bundled file of this sort often looks something like this:
+이러한 종류의 번들 파일은 종종 다음과 같은 코드를 가진다.
```
```suggestion
이러한 종류의 번들 파일은 종종 다음과 같이 작성할 수 있다.
```
아래와 같이 보인다...라고 해석하게 되면 확실히 이상해서 바꾸는 게 맞는 것 같은데요. '코드를 가진다'는 표현 보다 조금 더 자연스러울 수 있을 것 같아 제안드립니다...🙏

YeonjuOHYE committed:
전역 스코프는 정확히 어디에 존재하는가?

YeonjuOHYE committed:
Update ch4.md

YeonjuOHYE committed:
Update ch4.md

YeonjuOHYE committed:
오탈자

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -86,44 +86,45 @@ var moduleTwo = (function two(){
+이러한 파일들이 브라우저 환경에서 일반 독립 실행형 .js 파일로 별도로 로드되는 경우, 각 파일들의 최상위 변수 선언이 전역 변수가 된다. 왜냐하면 JS엔진에서는 전역 스코프만이 별개의 파일들(-독립적인 프로그램들)이 공유하는 유일한 리소스이기 떄문이다.
 
-In addition to (potentially) accounting for where an application's code resides during runtime, and how each piece is able to access the other pieces to cooperate, the global scope is also where:
+실행 시간 동안 애플리케이션의 코드가 존재하는 위치와 각 부분이 다른 부분에 접근하여 협력할 수 있는 방법을 설명하는 것 외에도, 전역 스코프는 다음에서도 찾아볼 수 있다:
```
`실행 시간 동안` -> `런타임 동안`으로 그대로 사용해도 될 것 같아요.

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -86,44 +86,45 @@ var moduleTwo = (function two(){
+전역 스코프가 파일의 가장 바깥쪽 부분, 즉 함수나 다른 블록 내부에 있지 않는 것 처럼 보일 수 있다. 하지만 그렇게 간단하지는 않다.
 
-Different JS environments handle the scopes of your programs, especially the global scope, differently. It's quite common for JS developers to harbor misconceptions without even realizing it.
+JS 환경마다 프로그램의 스코프, 특히 글로벌 스코프를 다르게 처리한다. JS 개발자이 자신도 모르는 사이에 잘못된 개념을 가지고 있는 것은 매우 흔한 일이다.
```
`JS 개발자이` -> `JS 개발자가`

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -173,19 +174,19 @@ console.log(window.something);
+`let` 선언은 `어떤` 전역 변수를 추가하지 전역 객체 속성은 추가하지 않는다(3장 참조). 그 효과는  `어떤` 렉시컬 식별자가 `어떤` 전역 객체 속성을 섀도잉하는 것이다.
 
-It's almost certainly a bad idea to create a divergence between the global object and the global scope. Readers of your code will almost certainly be tripped up.
+전역 객체와 전역 범위 간의 차이를 만드는 것은 확실히 좋지 않은 생각이다. 코드를 읽는 사람들은 거의 틀림없이 실수할 것이다.
```
`전역 범위` -> `전역 스코프`

wantyouring commented on scope-closures/ch4.md:
````diff
@@ -204,13 +205,13 @@ window["my-todo-list"];
 ```
 
-If the `id` value is a valid lexical name (like `first`), the lexical variable is created. If not, the only way to access that global is through the global object (`window[..]`).
+`id` 값이 유효한 렉시컬 이름(`first`와 같이)이면 렉시컬 변수가 생성햔다. 그렇지 않은 경우 전역 개체를 통해서만 전역에 접근할 수 있다(`window[..]`).    
````
`생성한다` -> `생성된다` 
또는
`렉시컬 변수가 생성한다` -> `렉시컬 변수를 생성한다`

가능한 능동표현이 좋다고 했어서 저는 후자가 좋을 것 같네요~

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -204,13 +205,13 @@ window["my-todo-list"];
+`id` 값이 유효한 렉시컬 이름(`first`와 같이)이면 렉시컬 변수가 생성햔다. 그렇지 않은 경우 전역 개체를 통해서만 전역에 접근할 수 있다(`window[..]`).    
 
-The auto-registration of all `id`-bearing DOM elements as global variables is an old legacy browser behavior that nevertheless must remain because so many old sites still rely on it. My advice is never to use these global variables, even though they will always be silently created.
+모든 `id`가 포함된 DOM 요소를 전역 변수로 자동 등록하는 것은 오래된 브라우저 레거시에며, 그럼에도 불구하고 많은 오래된 사이트들이 여전히 이 요소에 의존하고 있기 때문에 유지되어야 한다. 이러한 전역 변수가 생성되어있더라도 절대 사용하지 않는 것이 좋다.    
```
`레거시에며` -> `레거시이며`

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -219,23 +220,23 @@ console.log(name, typeof name);
+웹 워커는 기본 JS 프로그램을 실행하는 스레드와 완전히 다른 스레드(운영 체제 기준)에서 JS 파일을 실행할 수 있는 브라우저-JS 동작 위에 있는 웹 플랫폼 익스텐션이다.    
 
-Web Workers are a web platform extension on top of browser-JS behavior, which allows a JS file to run in a completely separate thread (operating system wise) from the thread that's running the main JS program.
+이러한 웹 워커 프로그램은 별도의 스레드에서 실행되기 때문에 애플리케이션의 메인 스레드와의 통신에서 제한되어 경합 조건 및 기타 문제를 방지/제한한다. 예를 들어 웹 워커 코드는 DOM에 대한 접근 권한이 없다. 그러나 `navigator`와 같이 일부 웹 API는 워커가 사용할 수 있다.    
```
`경합 조건` -> `경쟁 상태`

`race conditions`를 `경쟁 상태`로 많이 해석하는 것 같아요

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -219,23 +220,23 @@ console.log(name, typeof name);
 
-But the truly surprising behavior is that even though we assigned the number `42` to `name` (and thus `window.name`), when we then retrieve its value, it's a string `"42"`! In this case, the weirdness is because `name` is actually a pre-defined getter/setter on the `window` object, which insists on its value being a string value. Yikes!
+그런데 정말 놀라운 동작은 `42`라는 숫자를 `name`(따라서 `window.name`)에 붙였음에도 불구하고 그 값을 회수하면 `"42"`라는 문자열이 붙는다는 것이다. 이는 `window` 객체에 미리 정의된 getter/setter가 그 값을 string으로 만들기 때문이다. 으악!    
+    
```
여기 공백라인으로 아래 diff 일부가 밀려서 나오는 것 같아요

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -293,17 +294,17 @@ hello();
-If that code is in a file that's loaded as an ES module, it will still run exactly the same. However, the observable effects, from the overall application perspective, will be different.
-
-Despite being declared at the top level of the (module) file, in the outermost obvious scope, `studentName` and `hello` are not global variables. Instead, they are module-wide, or if you prefer, "module-global."
+ES 모듈로 로드된 파일에 해당 코드가 있으면 그대로 실행. 그러나 전체적인 적용 관점에서 관측 가능한 효과는 다를 것다.
```
`그대로 실행` -> `그대로 실행할 것이다`
`다를 것다` -> `다를 것이다` or `다르다`

`will`이 있기는 하지만 `다르다`라고 해도 될 것 같아요

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -293,17 +294,17 @@ hello();
+(모듈) 파일의 최상위 수준에서 선언되더라도 가장 바깥쪽의 명백한 스코프의 `studentName`과 `hello`는 전역 스코프가 아니다. 대신 모듈 범위<sub>module-wide</sub>의 또는 "module-global"이다.    
 
-However, in a module there's no implicit "module-wide scope object" for these top-level declarations to be added to as properties, as there is when declarations appear in the top-level of non-module JS files. This is not to say that global variables cannot exist or be accessed in such programs. It's just that global variables don't get *created* by declaring variables in the top-level scope of a  ...
+그러나 모듈에는 선언이 비모듈 JS 파일의 최상위 수준에 나타나는 것처럼 이러한 최상위 선언을 속성으로 추가할 내포된 "모듈 범위 스코프 개체"가 없다. 전역 변수가 존재하거나 이러한 프로그램에 접근 할 수 없다는 뜻은 아니다. 단지 모듈의 최상위 범위에서 변수를 선언한다고 전역 변수가 *생성*되는 것은 아니다.    
```
`전역 변수가 존재하거나 이러한 프로그램에 접근 할 수 없다는 뜻은 아니다.` 에서 `cannot exist or be accessed` 해석이 `전역 변수가 존재한다`는 쪽으로도 해석될 수 있을 것 같아요. 
`전역 변수가 존재할 수 없거나 이러한 프로그램에 접근할 수 없다는 뜻은 아니다.` 이런식으로 쓰는건 어떨까요?

YeonjuOHYE committed:
Update ch4.md

https://github.com/js-in-korean/You-Dont-Know-JS/pull/5#discussion_r677506376
https://github.com/js-in-korean/You-Dont-Know-JS/pull/5#discussion_r677516656
리뷰 반영

YeonjuOHYE committed:
7/28 리뷰 반영

YeonjuOHYE committed:
7/28 리뷰

YeonjuOHYE committed:
Update ch4.md

YeonjuOHYE committed:
Update ch4.md

YeonjuOHYE committed:
Update ch4.md

01kn commented on scope-closures/ch4.md:
```diff
@@ -408,14 +408,14 @@ const theGlobalScopeObject =
+휴! 이상적이진 않지만 신뢰할 수 있는 전역 객체 참조가 필요한 경우 유용하다.
 
-(The proposed name `globalThis` was fairly controversial while the feature was being added to JS. Specifically, I and many others felt the "this" reference in its name was misleading, since the reason you reference this object is to access to the global scope, never to access some sort of global/default `this` binding. There were many other names considered, but for a variety of reasons ruled out ...
+(제안된 이름 `globalThis`는 JS에 추가되는 동안 상당히 논란이 되었다. 특히, 당신이 이 객체를 참조하는 이유는 전역 스코프에 접근하기 위한 것이지 전역의/기본적인 `this` 바인딩에 접근하기 위한 것이 아니기 때문에 나와 다른 많은 사람들은 그 이름에 있는 "this" 참조가 오해의 소지가 있다고 느꼈다. 다른 많은 이름들이 고려되었지만, 여러 가지 이유로 제외되었다. 불행하게도, 선택된 이름은 최후의 수단이 되었다. 프로그램에서 전역 스코프 객체와 상호 작용할 계획이라면 혼란을 줄이기 위해 더 나은 이름을 선택할 것을 강력히 권장다. 여기서 (웃길 정도로 길지만 정확한!)'GlobalScopeObject'가 사용되었다.)
```
권장다. -> 권장한다.
'GlobalScopeObject' -> 'theGlobalScopeObject'

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -293,23 +293,23 @@ hello();
 ### Node
 
-One aspect of Node that often catches JS developers off-guard is that Node treats every single .js file that it loads, including the main one you start the Node process with, as a *module* (ES module or CommonJS module, see Chapter 8). The practical effect is that the top level of your Node programs **is never actually the global scope**, the way it is when loading a non-module file in the browse ...
+JS 개발자의 허를 찌르는 Node의 한 가지 측면은 Node 프로세스를 시작하는 기본 파일을 포함하여 로드되는 모든 .js 파일을 *module*(ES 또는 CommonJS 모듈, 8장 참조)로 처리한다는 것이다. 실제적인 효과는 브라우저에서 모듈이 아닌 파일을 로드할 때처럼 Node 프로그램의 최상위 레벨이 **전역 스코프**가 되는 것이 아니라는 것이다.    
```
`**전역 스코프가 되는 것이 아니라는 것이다.**`로 뒷 부분까지 볼드처리 되어야 할 것 같아요

wantyouring commented on scope-closures/ch4.md:
```diff
@@ -408,14 +408,14 @@ const theGlobalScopeObject =
+## 전역에 대한 이해
 
-The global scope is present and relevant in every JS program, even though modern patterns for organizing code into modules de-emphasizes much of the reliance on storing identifiers in that namespace.
+모든 JS 프로그램에 전역 스코프가 존재하며, 비록 코드를 모듈로 구성하기 위한 현대적 패턴이 해당 네임스페이스에 식별자를 저장하는 것에 대한 의존도를 크게 낮췄지만, 모든 JS 프로그램에 관련성이 있다.
```
비록 코드를 모듈로 구성하기 위한 현대적 패턴이 해당 네임스페이스에 식별자를 저장하는 것에 대한 의존도를 크게 낮췄지만, 모든 JS 프로그램에는 전역 스코프가 존재하고 관련성이 있다.

`even though` 이전 문장을 맨 뒤로 합쳐서 써도 좋을 것 같아요

sw-wm commented on scope-closures/ch4.md:
````diff
@@ -324,9 +324,9 @@ hello();
 ```
 
-Before processing, Node effectively wraps such code in a function, so that the `var` and `function` declarations are contained in that wrapping function's scope, **not** treated as global variables.
+프로세싱 전에, Node는 이러한 코드를 함수에 효과적으로 감싸서 'var' 및 'function' 선언이 전역 변수로 처리되지 않는 함수의 범위에 포함되도록 한다.    
````
```suggestion
프로세싱 전에, Node는 이러한 코드를 함수에 효과적으로 감싸서 'var' 및 'function' 선언이 전역 변수로 처리되지 **않는** 함수의 범위에 포함되도록 한다.
```
`not` 강조해야 할 것 같아요..!

YeonjuOHYE committed:
리뷰 반영

wantyouring reviewed (approved):
👍
